packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5536,6 +5556,10 @@ abstract class RenderObjectElement extends Element {
       _debugDoingBuild = false;
       return true;
     }());
+    assert(() {
+      Element.debugBuildRecorder?._recordRenderObjectElement(this);
```

Comment:
Why are ComponentElements only recorded during build, but RenderObjectElement also during update and mount?